### ISSUE TITLE
Update of Wookianos Web Font

### ISSUE THREAD

**Issue by dyacob** [keymanapp/keyboards#2072]
Keyman Web uses a web font (.woff) version of the Wookianos font that is referenced by many or most of the Ethiopic keyboards in the `.keyboard_info` files as per:

```
      "font": {
        "family": "GeezWeb",
        "source": [
          "wookianos.woff",
          "wookianos.ttf",
          "wookianos.mobileconfig",
          "ETHIOPI0.eot"
        ]
```

Wookianos, in turn, targets the Unicode 3.0 standard so is not suitable for a number of languages like Blin, Gurage and others. It also has ቬ misencoded in the position of ቭ which is only an occasional problem.   The Wookianos font was updated to support Gurage as per (Unicode 14) and Blin (as per Unicode 4.1) languages and can be found in the [gff_gurage](https://github.com/keymanapp/keyboards/tree/master/release/gff/gff_gurage) project in the `source` folder with the name `GurageEthiopicWookianos-Regular.ttf`.  Note that the Gurage fonts in the `gff_gurage` project in turn originate from the [gurage-font-collection](https://github.com/raeytype/gurage-font-collection) project.

Attached to this ticket is the same font, but renamed **"Keyman Wookianos"** , the archive includes `.woff` and .`woff2` web formats.  The refinements made are detailed in the  [FONTLOG-GurageEthiopicWookianos.txt](https://github.com/keymanapp/keyboards/blob/master/release/gff/gff_gurage/source/FONTLOG-GurageEthiopicWookianos.txt) in the `gff_gurage` project which can be repurposed as a FONTLOG file for "Keyman Wookianos".

I could not locate any of the files in the "GeezWeb" collection in keyman repositories.  If the "keyboards" project is the wrong place for this ticket, I can recreate the ticket in the appropriate repository if directed to it.

Food for thought, while the attached update to Wookianos is more complete than the original, it still does not provide every Ethiopic symbol defined in the Unicode standard.  The same I expect is true of the `ETHIOPI0.eot` font (which I could not locate).  It might be better to migrate to more modern and actively maintained fonts such as [Noto Ethiopic](https://github.com/notofonts/ethiopic/releases/tag/NotoSerifEthiopic-v2.102) or Abyssinica SIL which have full Unicode coverage for Ethiopic.

**Comment by dyacob** [comment]
I realized a short time later that there is one more facet to updating the Ethiopic web fonts worth considering.  The "Wookianos" font, and companions, are also in the [shared/fonts/geez](https://github.com/dyacob/keyboards/tree/master/release/shared/fonts/geez) folder.

I have also updated these companion fonts to support Gurage and Blin, but put them aside and got sidetracked with other things.  The updates are ready, I just need to rename them as the OFL requires.  I could simply add "GFF" as a prefix to their names, or "Keyman", or something else.  Does the Keyman team have a preference?

So I suggest now, making these updates together so they use consistent naming.  That is, update the fonts in the `shared/fonts/geez` folder, then create the .woff2 files, etc for Keyman Web.  Once updated in the `shared` folder, they can also be removed from the `gff_gurage` folder, and I'll go about updating the `.keyboard_info` file for all GFF keyboards.  I've planned to update them all to use multiple layers as per GFF Amharic 3.x.

**Comment by mcdurdin** [comment]
> ETHIOPI0.eot

The .eot format is deprecated (supported only in IE) so we no longer distribute that.

**Comment by mcdurdin** [comment]
> We probably only need to add the .woff2 file as according to @mcdurdin all browsers support woff2.

[woff2 support per caniuse.com](https://caniuse.com/?search=woff2)

**Comment by LornaSIL** [comment]
I guess I don't understand everything about this font. 
- The **Ethiopic Wookianos** (`wookianos.ttf`) font that is distributed with the .kmp files and also that is used for our help files (`s.keyman.com`) is v4.0
   - Copyright (c) 1998-2006 Senamirmir Project. All Right Reserved. Distributed under the terms of the GNU General Public License. As a special exception, if you create a document which uses this font, and embed this font or unaltered portions of this font into the document, this font does not by itself cause the resulting document to be covered by the GNU General Public License. This exception does not however invalidate any other reasons why the document might be covered by the GNU General Public License. If you modify this font, you may extend this exception to your version of the font, but you are not obligated to do so. If you do not wish to do so, delete this exception statement from your version.
- The **Keyman Ethiopic Wookianos** is v4.001 
   - Copyright (c) 2022 by Raey Type (https://github.com/raeytype), with Reserved Font Name "Keyman Ethiopic Wookianos".
Copyright (c) 1998-2008 Senamirmir Project. All Right Reserved.  Released under the SIL Open Font License; http://scripts.sil.org/OFL

Version 4.0 was under a GNU license according to the internals of the font. When was it released under the OFL?

If permission was given to change to OFL, what was the reserved font name? I think an RFN should just be separate words. And you shouldn't include "Keyman" or "Ethiopic" in an RFN as those are common words. You could make "Wookianos" an RFN, but did Senamirmir agree to that?

Lastly, is this font usable outside of Keyman? If so, why do we need "Keyman" in the font name?

**Comment by dyacob** [comment]
It seems that it was the case that the 4.0 version was GPL and it changed to OFL in 4.1.  I just double-checked and downloaded the Washra 4.1 ZIP archive from its homepage:  https://senamirmir.com/projects/typography/washra.html

Extracting the zip archive, the OFL.txt , OFL-FAQ.txt and a FONTLOG.txt file are all present.

The internal copyright statement for Wookianos font from the 4.1 zip file:

_Copyright (c) 1998-2008 Senamirmir Project. All Right Reserved.  Released under the SIL Open Font License; http://scripts.sil.org/OFL_

I do plan to get to the updates for the fonts in the "shared" directory as mentioned earlier, I'll use "GFF" as the name prefix, as OFL fonts they are usable outside of Keyman.  I thought to use "Keyman" initially when I thought the update would only be for the one Wookianos web font, but now I realize updates are needed for all of the regular true type fonts in the collection.

**Comment by LornaSIL** [comment]
Sounds good. I just want it to be an appropriate fix. So, your version would be bumped presumably. Maybe you could remove this .zip file so no one downloads it?

**Comment by dyacob** [comment]
I did remove the attachment.  I do recommend moving to Ethiopic Noto instead (Sans or Serif) as a web-based font. simply because it has the full coverage of all Ethiopic letters defined in Unicode.  The update that I have ready has better coverage, but is still incomplete.

**Comment by LornaSIL** [comment]
I am not seeing the new Gurage characters in the latest Noto Sans Ethiopic on the Google site. Can you give me a link to a more updated font?

**Comment by dyacob** [comment]
Version 2.102 , it's pretty recent and may not have been updated everywhere yet:

https://github.com/notofonts/ethiopic/releases/tag/NotoSerifEthiopic-v2.102

**Comment by LornaSIL** [comment]
I'm only adding Regular since we don't want to overwhelm our github repo or bloat the keyboard packages.

**Comment by LornaSIL** [comment]
You should be able to merge/master into your clone now. Then, feel free to update the `keyboard_info` files that include
```
      "font": {
        "family": "GeezWeb",
        "source": [
          "wookianos.woff",
          "wookianos.ttf",
          "wookianos.mobileconfig",
          "ETHIOPI0.eot"
        ]
```
I've added:
```
NotoSansEthiopic-Regular.ttf
NotoSansEthiopic-Regular.woff2
NotoSerifEthiopic-Regular.ttf
NotoSerifEthiopic-Regular.woff2
```
to the s.keyman.com repo which is where it draws the fonts for the help files and for KeymanWeb.
I didn't know whether you preferred sans or serif, so I added both...and both are 2.102 which include the Gurage glyphs.

**Comment by dyacob** [comment]
Thanks for generating the woff2 files @LornaSIL -I was just about to attach them here and see that you beat me to it by 5 days :-)    My personal preference is for the Serif style, but I think it is nice to have both available as options and the keyboard designer can select the one that he or she prefers.

**Comment by LornaSIL** [comment]
These are the `keyboard_info` files that call for `wookianos`
```
gff\gff_amharic\gff_amharic.keyboard_info
gff\gff_amharic_classic\gff_amharic_classic.keyboard_info
gff\gff_amh_7\gff_amh_7.keyboard_info
```
You could easily update those if you want without a version bump. However, if you also want to remove wookianos from the keyboard packages, these are the keyboards you should update:

```
gff\gff_amharic
gff\gff_amharic_classic
gff\gff_amh_7
gff\gff_gurage
gff\gff_tigrinya_eritrea
gff\gff_tigrinya_ethiopia
packages\gff_amh_powerpack_7
packages\gff_gurage_and_amharic
```

**Comment by dyacob** [comment]
Thanks @LornaSIL , I'm ready to give it a try with Amharic as a test, where the update would appear like:

```
      "font": {
        "family": "GeezWeb",
        "source": [
          "NotoSerifEthiopic-Regular.woff2",
          "NotoSerifEthiopic-Regular.ttf"
        ]
      },
```
Two questions.  (1) There is a `wookianos.mobileconfig` entry in the current `.keyboard_info` files.  Is a corresponding `NotoSerifEthiopic-Regular.mobileconfig` also needed?  I'm not familiar with what this file does.

(2) Also, is there a way to set `NotoSerifEthiopic` for regular web, and `NotoSansEthiopic` for mobile web?

thanks!

**Comment by LornaSIL** [comment]
You do not need a `.mobileconfig`. I think it's an outdated format.

Here is all I know (and more than I know):
https://help.keyman.com/developer/cloud/keyboard_info/1.0/#toc-examples

I do know you can set a font for `font` and another for `oskFont`.
And, I know this keyboard uses both `font` and `oskFont`:
https://keyman.com/keyboards/khmer_angkor

And the source:
https://github.com/keymanapp/keyboards/blob/master/release/k/khmer_angkor/khmer_angkor.keyboard_info

So, try it out and please tell me if one of them is for regular web and one for mobile web so I'll know too :)

**Comment by mcdurdin** [comment]
> Is a corresponding `NotoSerifEthiopic-Regular.mobileconfig` also needed? I'm not familiar with what this file does.

.mobileconfig files are an old way of bundling fonts for use on iOS. The iOS UX for this is super clunky so we don't really promote it. We [don't yet support font installation](https://github.com/keymanapp/keyman/issues/3445) on iOS with their more modern method.

**Comment by mcdurdin** [comment]
> (2) Also, is there a way to set `NotoSerifEthiopic` for regular web, and `NotoSansEthiopic` for mobile web?

No, not at this time. Font specifications are currently global. 

> I do know you can set a font for `font` and another for `oskFont`.

`font` specifies the display font used in text editing for apps that ask for that info (not many do this; Keyman itself does). `oskFont` specifies just the font on the on screen keyboard, where it differs from `font`. A good example of this is the Khmer Angkor keyboard which has the [Khmer Busra Kbd font](https://github.com/keymanapp/keyboards/pull/1710), so that keys with multiple diacritic characters, but without a base, display nicely.

**Comment by dyacob** [comment]
Thanks for the added explanation @mcdurdin .  I was wondering what `osk` was, I'll remember `k` for "keyboard".

With this better understanding, I'll leave #2102 the same, I think the Sans typeface will look best on the on screen keyboard (as used on hardware keyboards), we can reset it of course if it turns out bad.

**Comment by dyacob** [comment]
The test update for Amharic, ala #2102 , appears to have no unexpected side effects.  I'll go ahead and make the update for the remaining GFF keyboards (all in a single PR), then I believe this web font issue is fully addressed and this issue can be closed.

**Comment by LornaSIL** [comment]
@dyacob Can we close this PR?

**Comment by dyacob** [comment]
Yes :-)  I the web font migration is done for the GFF keyboards.

**Comment by LornaSIL** [comment]
Fixed by #2108
